lightningd/gossip_control.c

```diff
@@ -677,4 +679,86 @@ static const struct json_command dev_send_timestamp_filter = {
 	"Send {peerid} the timestamp filter {first} {range}"
 };
 AUTODATA(json_command, &dev_send_timestamp_filter);
+
+static void json_channel_range_reply(struct subd *gossip UNUSED, const u8 *reply,
+				     const int *fds UNUSED, struct command *cmd)
+{
+	struct json_result *response = new_json_result(cmd);
+	u32 final_first_block, final_num_blocks;
+	bool final_complete;
+	struct short_channel_id *scids;
+
+	if (!fromwire_gossip_query_channel_range_reply(tmpctx, reply,
+						       &final_first_block,
+						       &final_num_blocks,
+						       &final_complete,
+						       &scids)) {
+		command_fail(cmd, LIGHTNINGD,
+			     "Gossip gave bad gossip_query_channel_range_reply");
+		return;
+	}
+
+	if (final_num_blocks == 0 && final_num_blocks == 0 && !final_complete) {
+		command_fail(cmd, LIGHTNINGD,
+			     "Gossip refused to query peer");
+		return;
+	}
+
+	json_object_start(response, NULL);
+	json_add_num(response, "final_first_block", final_first_block);
```

Comment:
I don't understand what the `final_` prefix should convey, maybe that should be in the json doc?